ops/charm.py

```diff
@@ -1488,6 +1549,66 @@ def __init__(self, name: str, raw: Dict[str, Any]):
         self.type = raw['type']
 
 
+class DeviceType(enum.Enum):
+    """Types of device that may be requested."""
+    GPU = 'gpu'
+    NVIDIA_GPU = 'nvidia.com/gpu'
+    AMD_CPU = 'amd.com/gpu'
+
+
+class DeviceMeta:
```

Comment:
For the new types, I wonder if we should go all in on dataclasses (probably frozen), and add a `from_dict` method to parse from the dict. Kind of like how I did with `pebble.Notice` recently?

Reply:
@benhoyt further to this, what do you think about making the existing ones dataclasses as well, for more consistency? We can't really change the `__init__` signature (and so would need to also add custom `__repr__`) but then they would all have `__repr__`, `__eq__`, and (in 3.10+) `__match_args__`.

The older ones would have to not be frozen also, so that plus the inconsistent creation is not ideal, but it's still more consistent than having just having the most recently added types be dataclasses.

Reply:
Per discussion, yeah, it's not ideal -- but let's do the new types as dataclasses, but leave the other ones for now. If people are clamouring for them (I doubt it) we can always update those in a separate PR.